### ISSUE TITLE
Client to Client potential idea

### ISSUE THREAD

**Issue by MasterJames** [socketio/socket.io#2600]
I was hopeful that a Client to Client solution on top of socket io was not impossible.
I noted this Server negotiation stuff here.
https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API/Writing_WebSocket_servers

I've thought would it not be possible to have the server mitigate a more direct client to client connection by one or both of the clients having a server like port open that can make a separate secure handshake with each other like the server does?

Maybe someone can explain how it's not possible without changing the WebSocket spec. if so is there a plan or proposal for that there? Wherever there is? This is otherwise a proposal to discuss how socket io may be able to solve the client2client aka peer2peer double firewalled connection problem.

_[WebRTC seems overkill and not connecting as reliably with more strict (mobile provider) firewalls.
Is that true/confirm-able? or is that the better way to go in a p2p situation? How does Torrent do it differently? Does chrome remote desktop have to bounce everything off a server or it seems to achieve client to client connection as well, right?]_

Further R&D:
Basically thinking can you simple run a server connection from the client on the port that is opened by the first connection to the main server which also knows that port, which has therfore been opened by the firewall on the client side? or does it filter for only the server ip?
http://stackoverflow.com/questions/9687561/is-it-possible-to-set-up-a-socket-io-client-running-server-side-on-a-node-js-s

**Comment by darrachequesne** [comment]
That issue was closed automatically. Please check if your issue is fixed with the latest release, and reopen if needed (with a [fiddle](https://github.com/darrachequesne/socket.io-fiddle) reproducing the issue if possible).
